src/Mvc/Mvc.Core/test/ModelBinding/Binders/DictionaryModelBinderTest.cs

```diff
@@ -111,8 +112,8 @@ public static TheoryData<string, string, IDictionary<string, string>> StringToSt
 
             return new TheoryData<string, string, IDictionary<string, string>>
                 {
-                    { string.Empty, "[{0}]", dictionaryWithOne },
-                    { string.Empty, "[{0}]", dictionaryWithThree },
+                    { string.Empty, "[{0}]", dictionaryWithNone },
+                    { string.Empty, "[{0}]", dictionaryWithNone },
```

Comment:
```suggestion
                    { string.Empty, "[{0}]", dictionaryWithOne },
```

I think this a typo? We want to retain the test for `dictionaryWithOne`.

Reply:
And `dictionaryWithThree`!

Also, considering the issue focused on numeric keys, shouldn't we have at least one regression test with a numeric key?